src/drivers/wdm/queries/KeWaitLocal/KeWaitLocal.ql

```diff
@@ -1,15 +1,19 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 /**
+ * @id cpp/portedqueries/ke-wait-local

```

Comment:
Do we want to make this more granular/match the directory layout a bit? Maybe something like cpp/drivers/wdm/ke-wait-local? Eventually we'd like to get to the point that these aren't defined by being "ported" but rather by just being the definitive version of these checks, so I don't love putting "portedqueries" in the ID.

Reply:
I agree. I used what was already listed as the ID but it doesn't make sense to keep that portion. I'd rather skip the slashes to make it easier to target because the directory doesn't really matter. Lets use:
- PFxxxxx for PreFast ports we can use the the same digits as the CA ID
- SDVxxx for SDV ports, same digits as SDV rules
- QLxxx for CodeQL originals, starting at 001

Reply:
SDV doesn't have digits for its rules I don't think.  I would actually lean towards keeping references to the old PF/SDV names in the description or the qlhelp file, since eventually we want these queries to stand alone, and the query id having references to old tools in it seems like the wrong place to put that info.

For the slashes in the name, it doesn't have to match the directory layout 1:1, but I do think it would be nice to have the "drivers/" part at least to help group our queries together.  CodeQL does use slashes in some of its query IDs (i.e. https://github.com/github/codeql/blob/main/cpp/ql/src/PointsTo/PreparedStagedPointsTo.ql).  @raulgarciamsft, do you have thoughts? You probably have more experience with query ID conventions than we do.

Reply:
Query ID and query Name are separate by the way, if you didn't realize that.
Queries should be grouped by function and not by name or location; if we rename a directory or move something around it would mess up your suggested system. I'm comfortable referencing the IDs from previous engines in the qlhelp file and as a note in the metadata in order to maintain reference/connection with the old tools.
As far as ID goes, I stick with my suggestion above, and that the SDV rules start at SDV001 since they don't have digits in their IDs, or that all the rules become SDATxxx or WDxxx (Windows Driver) or something of the sort.

Reply:
Yeah, my bad, I should've said slashes in the ID.  

I'm not saying we should name the ID after the directory, just that putting "/drivers/" in the id makes them easy to group separately from standard C/CPP rules.  

Can you share an example of what you think the IDs should look like, with the numbering? From what I've seen most CodeQL IDs are semantic rather than numbered (though again, I'd love to get Raul's input to better understand the conventions used here and in ECA.) It looks like we can also use an "@opaqueid" metadata field to make a distinction between the "human-readable" ID and a more unique/numeric identifier.

Reply:
As per Nate, Raul, and my conversation offline, we'll be moving forward with the standard ID format of: cpp/drivers/<light description>